source-map.bs

```diff
@@ -274,6 +290,44 @@ fields represent mapped code that also has a mapped name.
 Note: Using file offsets was considered but rejected in favor of using line/column data to avoid becoming
 misaligned with the original due to platform-specific line endings. 
 
+### [=names=] for generated JavaScript code
+
+Source map generators should create a [=mapping=] entry with a [=name=] field for
+the following JavaScript tokens, if
+
+1. The original source language construct maps semantically to the generated
+    JavaScript code.
+
+1. The original source language construct has a name.
+
+Then the [=name=] of the [=mapping=] entry should be the name of the original
+source language construct.
+
+Example: A TypeScript arrow function compiled to a JavaScript function, or a
```

Comment:
Is an arrow function a good example? Arrow functions don't have names unless they are assigned names, i.e., `let f = () => {}`. It isn't clear to me if this section intends for a source map generator to emit the name `f` for that function.

Reply:
The function in this case has name `f`. You can test it by running
```js
let f = () => {};
console.log(f.name);
```

Reply:
> The function in this case has name `f`. You can test it by running

Yes, I mentioned assigned names in my initial comment. To clarify, I'm asking if this section should more clearly spell out that a name could be defined for an arrow function for cases such as this.

Reply:
Yeah I think the other way makes more sense as an example: "A TypeScript function compiled to a JavaScript arrow function, ..."